osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderSelectionBlueprint.cs

```diff
@@ -626,10 +626,33 @@ private void convertToStream()
         public override Vector2 ScreenSpaceSelectionPoint => DrawableObject.SliderBody?.ToScreenSpace(DrawableObject.SliderBody.PathOffset)
                                                              ?? BodyPiece.ToScreenSpace(BodyPiece.PathStartLocation);
 
-        protected override Vector2[] ScreenSpaceAdditionalNodes => new[]
-        {
+        protected override Vector2[] ScreenSpaceAdditionalNodes => getScreenSpaceControlPointNodes().Prepend(
             DrawableObject.SliderBody?.ToScreenSpace(DrawableObject.SliderBody.PathEndOffset) ?? BodyPiece.ToScreenSpace(BodyPiece.PathEndLocation)
-        };
+        ).ToArray();
+
+        private IEnumerable<Vector2> getScreenSpaceControlPointNodes()
+        {
+            // Returns the positions of control points that produce visible kinks on the slider's path
+            // This excludes inherited control points from Bezier, B-Spline, Perfect, and Catmull curves
+            if (DrawableObject.SliderBody == null)
+                yield break;
+
+            PathType? currentPathType = DrawableObject.HitObject.Path.ControlPoints.FirstOrDefault()?.Type;
+
+            // Skip the first control point because it is already covered by the slider head
+            // Skip the last control point because its always either not on the slider path or exactly on the slider end
+            foreach (var controlPoint in DrawableObject.HitObject.Path.ControlPoints.Skip(1).SkipLast(1))
```

Comment:
Rather than issue two separate linq calls here, just `for (int i = 1; i < DrawableObject.HitObject.Path.ControlPoints.Count - 1; i++)`...?

Yeah this gets more complicated with the "current path type" accounting, but you can pretty easily get past that too (change for bounds to `0...Count - 1`, do the current segment type account first, and then `continue` on `i == 0 || i == Count - 1` before determining if the point is a "visible kink").